flink-end-to-end-tests/test-scripts/test_ha.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+################################################################################
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+source "$(dirname "$0")"/common.sh
+
+#FLINK_DIR=/Users/kkloudas/repos/dataartisans/flink/build-target flink-end-to-end-tests/test-scripts/test_ha.sh
+TEST_PROGRAM_JAR=$TEST_INFRA_DIR/../../flink-examples/flink-examples-streaming/target/StateMachineExample.jar\ --error-rate\ 0.0\ --sleep\ 2
+
+stop_cluster_and_watchdog() {
+    kill ${watchdogPid} 2> /dev/null
+    wait ${watchdogPid} 2> /dev/null
+
+    stop_ha_cluster
+}
+
+verify_logs() {
+    expectedRetries=$1
+
+    # verify that we have no alerts
+    if ! [ `cat ${output} | wc -l` -eq 0 ]; then
+        echo "FAILURE: Alerts found at the StateMachineExample with 0.0 error rate."
+        PASS=""
```

Comment:
What is `PASS` being used for?

Reply:
It is used in the `check_all_pass` in the `common.sh`. Kind of obscure, but this is the structure for now...